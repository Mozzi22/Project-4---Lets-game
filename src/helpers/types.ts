export type TRegBody = { login: string, password: string, confirm: string };
export type TLoginBody = { login: string, password: string };
