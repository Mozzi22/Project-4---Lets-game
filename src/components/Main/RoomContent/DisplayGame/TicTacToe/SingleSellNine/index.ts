import SingleSellNine from './SingleSellNine';

export default SingleSellNine;
