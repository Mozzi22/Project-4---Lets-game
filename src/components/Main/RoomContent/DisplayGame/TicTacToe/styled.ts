import styled from 'styled-components';

export const StTicTacToe = styled.div`
    background: white;
    width: 399px;
    height: 399px;
    display: flex;
    align-content: center;
    flex-wrap: wrap;
    flex-direction: row-reverse;
    color:red;
    margin: 10px auto;
`;
