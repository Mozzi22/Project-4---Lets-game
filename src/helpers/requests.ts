import { routes } from '../constants/routes';
import { TUrl } from './types';

const { baseUrl } = routes;

export const getRequest = async (url: TUrl) => {
  const options = { method: 'GET' };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};

export const postRequest = async (url: TUrl, body) => {
  const options = {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(body),
  };
  const answer = await fetch(`${baseUrl}${url}`, options);
  return answer;
};
