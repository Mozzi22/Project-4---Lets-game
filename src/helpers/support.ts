export const support = {
    setSessionStorageItem: (name: string, data: string) => {
        if (typeof data !== 'string') data = JSON.stringify(data);
        localStorage.setItem(name, data);
    },
    killSessionStorageItem: (name: string) => {
        localStorage.removeItem(name);
    },
    getSessionStorageItem: (name: string) => {
        const item = localStorage.getItem(name);
        return item?.charAt(0) === '{' ? JSON.parse(item) : item;
    },
};
