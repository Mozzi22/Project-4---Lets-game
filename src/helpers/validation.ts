import { regExp } from '../constants/regExp';
import { TRegBody, TLoginBody } from './types';

export const validation = {
    registrationValidation: ({ login, password, confirm }: TRegBody) => {
        if (!regExp.loginRegExp.test(login) || !login) return { message: 'invalid_login', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) return { message: 'invalid_password', isValid: false };
        if (password !== confirm) return { message: 'password_mismatch', isValid: false };
        return { message: '', isValid: true };
    },
    loginValidation: ({ login, password }: TLoginBody) => {
        if (!regExp.loginRegExp.test(login) || !login) return { message: 'invalid_login', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) return { message: 'invalid_password', isValid: false };
        return { message: '', isValid: true };
    },
};
