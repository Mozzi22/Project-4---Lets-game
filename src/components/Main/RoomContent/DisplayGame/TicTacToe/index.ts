import { connect } from 'react-redux';
import { getTicTacToeDesk } from 'src/Store/games/selectors';
import { do_step } from 'src/Store/games/actions';
// import { AplicationState } from 'src/Store/types';
import TicTacToe from './TicTacToe';

const mapStateToProps = (state) => ({
    ticTacToe: getTicTacToeDesk(state),
});

const mapDispatchToProps = {
  do_step,
};

export default connect(mapStateToProps, mapDispatchToProps)(TicTacToe);
