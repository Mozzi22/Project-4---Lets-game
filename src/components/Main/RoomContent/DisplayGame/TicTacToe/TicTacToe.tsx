import React from 'react';
import { StTicTacToe } from './styled';
import SingleSellNine from './SingleSellNine';

const TicTacToe = ({ ticTacToe, doStepTicTacToe }) => {

    return (
        <StTicTacToe>
           {ticTacToe.map((item) =>
                <SingleSellNine
                    key={item.position}
                    id={item.position}
                    figure={item.figure}
                    isChecked={item.isChecked}
                    onClick={doStepTicTacToe}
                />
            )}
        </StTicTacToe>
    );
};

export default TicTacToe;
