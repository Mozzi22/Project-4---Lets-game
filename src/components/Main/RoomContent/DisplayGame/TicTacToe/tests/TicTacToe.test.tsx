import React from 'react';
import TicTacToe from '../TicTacToe';
import { mountSmart } from 'src/helpers/testHelper';
import { mount, shallow } from 'enzyme';

describe('TicTacToe', () => {
    const props = {
        ticTacToe: [{
            position: 1,
            isChecked: true,
            figure: 'O'
        }],
        doStepTicTacToe: jest.fn(),
   }
    it('Should match snapshot', () => {
        const component = shallow(<TicTacToe {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StTicTacToe', () => {
        const component = mount(<TicTacToe {...props}/>);
        expect(component.find('styled__StTicTacToe')).toHaveLength(1);
    });
    it('should render SingleSellNine', () => {
        const component = mount(<TicTacToe {...props}/>);
        expect(component.find('SingleSellNine')).toHaveLength(1);
    });
    it('should click on the button', () => {
        const component = mountSmart(<TicTacToe {...props} />);
        component.find('SingleSellNine').props().onClick();
        expect(props.doStepTicTacToe).toHaveBeenCalled();
    });
});
