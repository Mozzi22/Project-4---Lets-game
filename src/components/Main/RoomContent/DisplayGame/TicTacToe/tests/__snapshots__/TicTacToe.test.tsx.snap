// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicTacToe Should match snapshot 1`] = `"<div class=\\"styled__StTicTacToe-sc-3am6n8-0 bMwVWP\\"><div class=\\"styled__StSingleSellNine-sc-1icsnrc-0 fXpGzf\\"><div class=\\"styled__StCrossAndZero-sc-1e5txb5-0 bWPZNh\\">O</div></div></div>"`;
